tests/browser_context_test.go

```diff
@@ -688,3 +689,156 @@ func TestBrowserContextTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestBrowserContextWaitForEvent(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name            string
+		event           string
+		optsOrPredicate *optsOrPredicate
+		wantErr         string
+	}{
+		{
+			name:  "successfully wait for page creation",
+			event: "page",
+		},
+		{
+			name:            "successfully wait for page creation with only predicate",
+			event:           "page",
+			optsOrPredicate: &optsOrPredicate{justPredicate: stringPtr("() => true;")},
+		},
+		{
+			name:            "successfully wait for page creation with predicate in option",
+			event:           "page",
+			optsOrPredicate: &optsOrPredicate{predicate: stringPtr("() => true;")},
+		},
+		{
+			name:            "successfully wait for page creation with predicate and timeout in option",
+			event:           "page",
+			optsOrPredicate: &optsOrPredicate{predicate: stringPtr("() => true;"), timeout: int64Ptr(1000)},
+		},
+		{
+			name:    "fails when event other than page passed in",
+			event:   "browser",
+			wantErr: "\"page\" is the only event that is supported, you passed in \"browser\"",
+		},
+		{
+			name:            "fails due to timeout",
+			event:           "page",
+			optsOrPredicate: &optsOrPredicate{predicate: stringPtr("() => false;"), timeout: int64Ptr(10)},
+			wantErr:         "waitForEvent timed out after 10ms",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tb := newTestBrowser(t)
+
+			bc, err := tb.NewContext(nil)
+			require.NoError(t, err)
+
+			ctx, cancel := context.WithTimeout(tb.context(), 5*time.Second)
+			defer cancel()
+
+			var p1, p2 api.Page
+			err = tb.run(
```

Comment:
It would be great if you could comment on what this code does. A simple comment, like: "Wait until the page is created". By the way, what would happen if the page is created before `WaitForEvent` is called? Is it expected?

Reply:
Sure, i've added some comments in 8fd96c5507cd0d0f1d86f00b498123f08475d56d. Let me know if they help or not.

Reply:
Thanks! It looks good.

I would remove the self-explanatory parts like:
- "// Call waitForEvent."
- "// Call newPage."
- "// For the happy paths."
- "// For the failure cases."

It's fine to keep only the ones that include whys. Also, the func comments you added are nice 👍

Reply:
Resolved in a fixup: ee71432cc301e6b3a0f36a809fc0e0f09316375a